src/plugins_exts/schema_mount.c

```diff
@@ -717,29 +717,29 @@ lyplg_ext_schema_mount_get_parent_ref(const struct lysc_ext_instance *ext, struc
 
         term = (struct lyd_node_term *)pref_set->dnodes[i];
         LYD_VALUE_GET(&term->value, xp_val);
-        LY_CHECK_GOTO(res = lyplg_type_print_xpath10_value(xp_val, LY_VALUE_JSON, NULL, &value, &err), out);
-        LY_CHECK_ERR_GOTO(res = lys_find_xpath(ext->module->ctx, NULL, value, 0, &snode_set), free(value), out);
+        LY_CHECK_GOTO(res = lyplg_type_print_xpath10_value(xp_val, LY_VALUE_JSON, NULL, &value, &err), error);
+        LY_CHECK_ERR_GOTO(res = lys_find_xpath(ext->module->ctx, NULL, value, 0, &snode_set), free(value), error);
         free(value);
         for (uint32_t sn = 0; sn < snode_set->count; sn++) {
-            struct lysc_node *snode = snode_set->snodes[sn];
-
-            if ((res = ly_set_add(results_set, snode, 0, NULL))) {
-                ly_set_free(snode_set, NULL);
-                ly_set_free(results_set, NULL);
-                goto out;
-            }
+            LY_CHECK_GOTO(res = ly_set_add(results_set, snode_set->snodes[sn], 0, NULL), error);
         }
         ly_set_free(snode_set, NULL);
+        snode_set = NULL;
     }
 
     *refs = results_set;
 
 out:
+    ly_set_free(snode_set, NULL);
     if (ext_data_free) {
         lyd_free_all(ext_data);
     }
     ly_set_free(pref_set, NULL);
     return res;
+
+error:
+    ly_set_free(results_set, NULL);
+    goto out;
 }
```

Comment:
Merge into `cleanup` and start it with `if (res) { ly_set_free(results_set, NULL); }`.